add_slide.php

```diff
@@ -0,0 +1,177 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Add slide form for carousel
+ *
+ * @package     block_carousel
+ * @copyright   Peter Burnett <peterburnett@catalyst-au.net>
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+require_once(__DIR__ . '/../../config.php');
+require_once($CFG->libdir . '/filelib.php');
+$blockid = required_param('bid', PARAM_INT);
+$action = optional_param('action', '', PARAM_TEXT);
+$id = optional_param('id', 0, PARAM_INT);
+
+$context = context_block::instance($blockid);
+$PAGE->set_context(context_block::instance($blockid));
+$PAGE->set_title(get_string('addslide', 'block_carousel'));
+$PAGE->set_heading(get_string('addslide', 'block_carousel'));
+$url = new moodle_url('/blocks/carousel/add_slide.php', ['bid' => $blockid, 'action' => $action, 'id' => $id]);
+$PAGE->set_url($url);
+
+require_login();
+require_capability('moodle/block:edit', $context);
+
+$editurl = "carousel_{$blockid}_editurl";
+$prevurl = $SESSION->$editurl;
```

Comment:
I can't see this being used anywhere?

Reply:
its used in all the page actions that return to the prev page. Eg clone, delete, enable, disable